src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -173,58 +171,57 @@ class AuthScreens extends React.Component {
     }
 
     render() {
-        const commonModalScreenOptions = {
+        const commonScreenOptions = {
             headerShown: false,
             gestureDirection: 'horizontal',
             animationEnabled: true,
-
-            // This option is required to make previous screen visible underneath the modal screen
-            // https://reactnavigation.org/docs/6.x/stack-navigator#transparent-modals
-            presentation: 'transparentModal',
-        };
-        const modalScreenOptions = {
-            ...commonModalScreenOptions,
-            cardStyle: getNavigationModalCardStyle({
-                windowHeight: this.props.windowHeight,
-                isSmallScreenWidth: this.props.isSmallScreenWidth,
-            }),
             cardStyleInterpolator: (props) => modalCardStyleInterpolator(this.props.isSmallScreenWidth, false, props),
             cardOverlayEnabled: true,
+            animationTypeForReplace: 'push',
+        };
 
-            // This is a custom prop we are passing to custom navigator so that we will know to add a Pressable overlay
-            // when displaying a modal. This allows us to dismiss by clicking outside on web / large screens.
-            isModal: true,
+        const rightModalNavigatorScreenOptions = {
+            ...commonScreenOptions,
+            // we want pop in RHP since there are some flows that would work weird otherwise
+            animationTypeForReplace: 'pop',
+            cardStyle: styles.navigationModalCard(this.props.isSmallScreenWidth),
```

Comment:
This caused a regression https://github.com/Expensify/App/issues/20709. We should have kept using `getNavigationModalCardStyle`. We modified this function in https://github.com/Expensify/App/pull/19682 but that change was not caught after we merged `main` https://github.com/Expensify/App/pull/19263/commits/24d3f342aae426725bd6f33b4c9fab87381e29bb#diff-5b711055e18a2a6ec042cebe121391e826554644d2965d3365250d90d5d39c32